### ISSUE TITLE
PLE4703 false positive for loop statements with break

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#11695]
```python
my_set = {1, 2, 3, 4}
for a in my_set:
    if a == "2":
        my_set.remove(a)
        break
```
runs fine, but is flagged by PLE4703. If the break statement was not here, it would indeed be an error. However, since it is added after the remove statement before the iterator is accessed again, it is valid. 

Still an issue on `ruff 0.4.7`

**Comment by augustelalande** [comment]
Not necessarily saying PLE4703 shouldn't be changed, but is there a *useful* example where this comes up?

The construction above should really be rewritten as
```python
my_set.discard("2")
```

**Comment by AlexWaygood** [comment]
Thanks for the report! Looks like it was already raised in https://github.com/astral-sh/ruff/issues/10721, however, so I'll fold this into that one :-)
